stake-pool/cli/src/main.rs

```diff
@@ -562,6 +562,104 @@ fn command_withdraw(
     Ok(None)
 }
 
+fn command_set_staking_auth(
+    config: &Config,
+    pool: &Pubkey,
+    stake_account: &Pubkey,
+    new_staker: &Pubkey,
+) -> CommandResult {
+    let pool_data = config.rpc_client.get_account_data(&pool)?;
+    let pool_data: StakePool = PoolState::deserialize(pool_data.as_slice())
+        .unwrap()
+        .stake_pool()
+        .unwrap();
+
+    let pool_withdraw_authority: Pubkey = PoolProcessor::authority_id(
+        &spl_stake_pool::id(),
+        pool,
+        PoolProcessor::AUTHORITY_WITHDRAW,
+        pool_data.withdraw_bump_seed,
+    )
+    .unwrap();
+
+    let mut transaction = Transaction::new_with_payer(
+        &[set_staking_authority(
+            &spl_stake_pool::id(),
+            &pool,
+            &config.owner.pubkey(),
+            &pool_withdraw_authority,
+            &stake_account,
+            &new_staker,
+            &stake_program_id(),
+        )?],
+        Some(&config.fee_payer.pubkey()),
+    );
+
+    let (recent_blockhash, fee_calculator) = config.rpc_client.get_recent_blockhash()?;
+    check_fee_payer_balance(config, fee_calculator.calculate_fee(&transaction.message()))?;
+    let mut signers = vec![config.fee_payer.as_ref(), config.owner.as_ref()];
+    unique_signers!(signers);
+    transaction.sign(&signers, recent_blockhash);
+    Ok(Some(transaction))
+}
+
+fn command_set_owner(
+    config: &Config,
+    pool: &Pubkey,
+    new_owner: &Option<Pubkey>,
+    new_fee_receiver: &Option<Pubkey>,
+) -> CommandResult {
+    let pool_data = config.rpc_client.get_account_data(&pool)?;
+    let pool_data: StakePool = PoolState::deserialize(pool_data.as_slice())
+        .unwrap()
+        .stake_pool()
+        .unwrap();
+
+    // If new accounts are missing in the arguments use the old ones
+    let new_owner: Pubkey = match new_owner {
+        None => pool_data.owner,
+        Some(value) => *value,
+    };
+    let new_fee_receiver: Pubkey = match new_fee_receiver {
+        None => pool_data.owner,
```

Comment:
Be careful here -- the `new_fee_receiver` is meant to be a new fee account, so this should be `pool_data.owner_fee_account`.  I looked back at `process_set_owner`, and we don't do any checks on the new fee account.  We should add a check that it's actually a valid pool token account to avoid causing a problem later.

Reply:
We do have a check for the new fee account to have a proper mint assigned to it.
Using owner as default is nasty, great catch.

Reply:
Yes true, you're doing it here, which is great.  We also need a check at https://github.com/solana-labs/solana-program-library/blob/master/stake-pool/program/src/processor.rs#L561 though -- could you add it with this PR?

Reply:
Ok, got it, I guess we should add it to the stake pool initialization as well.

Reply:
Yes, please do

Reply:
Done :)